projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-site-logo-endpoint.php

```diff
@@ -68,29 +66,29 @@ function callback( $path = '', $site_id = 0 ) {
 
 		$args = $this->input();
 		$logo_settings = $this->get_current_settings();
+
 		if ( empty( $args ) || ! is_array( $args ) ) {
 			return $logo_settings;
 		}
 
 		if ( isset( $args['id'] ) ) {
-			$logo_settings['id'] = (int) $args['id'];
-		}
-		if ( isset( $args['url'] ) ) {
-			$logo_settings['url'] = $args['url'];
-		}
-		if ( isset( $args['url'] ) || isset( $args['id'] ) ) {
-			update_option( 'site_logo', $logo_settings );
+			update_option( 'site_logo', (int) $args['id'] );
 		}
 
 		return $this->get_current_settings();
 	}
 
 	function get_current_settings() {
-		$logo_settings = get_option( 'site_logo' );
-		if ( ! is_array( $logo_settings ) ) {
-			$logo_settings = array();
+		$logo_id = get_option( 'site_logo' );
+
+		if ( ! $logo_id ) {
+			return array();
 		}
-		return $logo_settings;
+
+		return array(
```

Comment:
Same question as posted [here](https://github.com/Automattic/jetpack/pull/20192#discussion_r661764922). Do we need to update the `sizes` value here?